vaccine_feed_ingest/apis/placekey.py

```diff
@@ -91,43 +95,58 @@ def lookup_placekeys(
                 ],
             )
 
-            cache_response = self.api_cache.get(cache_key)
+            cache_response = self.api_cache.get(cache_keys[record_id])
 
             if cache_response:
-                if "error" in cache_response:
-                    result[record_id] = None
-                    continue
-
-                result[record_id] = cache_response.get("placekey")
+                if "error" not in cache_response and "placekey" in cache_response:
+                    result[record_id] = cache_response["placekey"]
                 continue
 
-            response = self._placekey_api.lookup_placekey(
-                latitude=latitude,
-                longitude=longitude,
-                location_name=location_name,
-                street_address=street_address,
-                city=city,
-                region=region,
-                postal_code=postal_code,
-                iso_country_code=iso_country_code,
-                strict_address_match=strict_address_match,
-                strict_name_match=strict_name_match,
+            places.append(
+                {
+                    "query_id": record_id,
+                    "latitude": latitude,
+                    "longitude": longitude,
+                    "location_name": location_name,
+                    "street_address": street_address,
+                    "city": city,
+                    "region": region,
+                    "postal_code": postal_code,
+                    "iso_country_code": iso_country_code,
+                }
             )
 
+        # Bulk load remaining places from API
+        if not places:
+            return result
+
+        responses = self._placekey_api.lookup_placekeys(
+            places,
+            strict_address_match=strict_address_match,
+            strict_name_match=strict_name_match,
+        )
+
+        if not responses:
+            return result
+
+        for response in responses:
+            record_id = response.pop("query_id")
+
             if not response:
-                result[record_id] = None
                 continue
 
             if "error" in response:
                 logger.info("Failed to add placekey because: %s", response["error"])
-                self.set_with_expire(cache_key, {"error": response["error"]})
-                result[record_id] = None
+                self.set_with_expire(
+                    cache_keys[record_id], {"error": response["error"]}
```

Comment:
Should we set a _shorter_ cache time for errors, or do we expect them to be persistent?

From a local run it looks like the errors I see are all related to the source data (e.g., bad address, which will be handled by the cache key).

Reply:
I don't want a lower cache time unless it is temporary error on the server side, which I don't really have a hook to know when there is a temporary error.

I definetly want to cache when the address is bad so we don't continue to call placekey for bad addresses.